common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -331,13 +344,13 @@ static MergedShuffleFileManager newMergedShuffleFileManagerInstance(
   }
 
   private void loadSecretsFromDb() throws IOException {
-    secretsFile = initRecoveryDb(SECRETS_RECOVERY_FILE_NAME);
+    secretsFile = initRecoveryDb(SECRETS_RECOVERY_FILE_NAME + dbBackend.suffix());
```

Comment:
minor nit: perhaps we can add `dbBackend.fileName(prefix: String)` or some such instead  (and remove suffix as well) ?

Reply:
Good to me, [9590e97](https://github.com/apache/spark/pull/36200/commits/9590e976bf330de20ff2a33b6ecb1810a2bf5422) fix this